cfme/tests/cloud/test_project_quota.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+from widgetastic.utils import partial_match
+
+from cfme import test_requirements
+from cfme.cloud.instance import Instance
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.generators import random_vm_name
+
+pytestmark = [
+    test_requirements.quota,
+    pytest.mark.long_running,
+    pytest.mark.provider([AzureProvider, OpenStackProvider],
+                         required_fields=[['provisioning', 'image']], scope="module")
+]
+
+
+@pytest.fixture
+def vm_name():
+    return random_vm_name(context='quota')
+
+
+@pytest.fixture
+def template_name(provisioning):
+    return provisioning["image"]["name"]
+
+
+@pytest.fixture(scope="module")
+def roottenant(appliance):
+    return appliance.collections.tenants.get_root_tenant()
+
+
+@pytest.fixture
+def prov_data(provider, provisioning, vm_name, template_name):
+    if provider.one_of(OpenStackProvider):
+        return {
+            "catalog": {'vm_name': vm_name, 'catalog_name': {'name': template_name}},
+            "environment": {'automatic_placement': True},
+            "properties": {'instance_type': partial_match('m1.large')}
+        }
+    if provider.one_of(AzureProvider):
+        return {
+            "catalog": {'vm_name': vm_name, 'catalog_name': {'name': template_name}},
+            "environment": {'automatic_placement': True},
+            "properties": {'instance_type': partial_match('d2s_v3')},
+            "customize": {'admin_username': provisioning['customize_username'],
+                          'root_password': provisioning['customize_password']}
+        }
+
+
+@pytest.fixture
+def set_project_quota(request, appliance, new_project):
```

Comment:
This logic to set quota seems to be identical in most places, and all that changes is arg #3. Perhaps this logic should live in a common helper method used by both tests?